types/globalthis/index.d.ts

```diff
@@ -0,0 +1,29 @@
+// Type definitions for globalthis 1.0
+// Project: https://github.com/ljharb/System.global#readme
+// Definitions by: ExE Boss <https://github.com/ExE-Boss>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.4
```

Comment:
Support for&nbsp;`globalThis` was added in&nbsp;[**TypeScript&nbsp;3.4**](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#type-checking-for-globalthis).

Reply:
it’s the same type as `windows or `global`; were these types not in earlier versions?

Reply:
There is&nbsp;`declare var window`, but&nbsp;that&nbsp;requires `lib: ["dom"]`, and&nbsp;`declare var global`, which&nbsp;requires&nbsp;`@types/node`, but&nbsp;only the&nbsp;former one&nbsp;has, and&nbsp;this&nbsp;only exists in&nbsp;**TypeScript&nbsp;3.4+**:
```ts
declare var window: Window & typeof globalThis;
declare var self: Window & typeof globalThis;
```

Node’s `global` only has:
```ts
declare var global: NodeJS.Global;
```

---

Also, even if you do:
```ts
var something;
```

Even though it&nbsp;will be&nbsp;available as&nbsp;`window.something` during&nbsp;runtime, without&nbsp;`globalThis`, the&nbsp;compiler wouldn’t&nbsp;know&nbsp;that.

---

Also, in&nbsp;`lib.dom.d.ts`, `Window`&nbsp;functions have&nbsp;to&nbsp;duplicated in&nbsp;the&nbsp;global&nbsp;scope, so&nbsp;that&nbsp;you&nbsp;can&nbsp;do:
```ts
addEventListener("foo", e => {
	// Process the `foo` event
})
```

Same&nbsp;with&nbsp;Node&nbsp;globals.